### ISSUE TITLE
double "Transfer-Encoding" headers

### ISSUE THREAD

**Issue by diasdavid** [ipfs/go-ipfs#2385]
caught by @RichardLitt documentation efforts:

https://github.com/ipfs/http-api-spec/blob/master/apiary.apib#L2938-L2942

**Comment by dignifiedquire** [comment]
So I found out a bit more about this. According to the `http` spec these are equivalent:

```
HTTP/1.1 200 OK\r\n
Transfer-Encoding: chunked\r\n
Transfer-Encoding: chunked\r\n
```
should be the same as

```
HTTP/1.1 200 OK\r\n
Transfer-Encoding: chunked, chunked\r\n
```

So the question is, are we actually double chunking our responses, and if so why?

This is also making some problems for me as I'm trying to use rust http libraries which are not yet able to handle this:

* https://github.com/tailhook/rotor-http/issues/34
* https://github.com/hyperium/hyper/issues/683

**Comment by whyrusleeping** [comment]
Okay, so we're sending the transfer encoding twice... which is wrong. but we arent doing double chunked encoding. I'll take a look

**Comment by Kubuxu** [comment]
Resolved by #2465. If not please reopen.
